src/iop/colorout.c

```diff
@@ -804,6 +842,36 @@ void commit_params(struct dt_iop_module_t *self, dt_iop_params_t *p1, dt_dev_pix
   dt_ioppr_set_pipe_output_profile_info(self->dev, piece->pipe, d->type, out_filename, p->intent);
 }
 
+void modify_roi_out(dt_iop_module_t *self,
+                    dt_dev_pixelpipe_iop_t *piece,
+                    dt_iop_roi_t *roi_out,
+                    const dt_iop_roi_t *const roi_in)
+{
+  *roi_out = *roi_in;
+  roi_out->x = MAX(0, roi_in->x);
+  roi_out->y = MAX(0, roi_in->y);
+}
```

Comment:
It would be useful for future maintainers if you added the descriptions for the modify\_\* functions as found in e.g. clipping.c.

Reply:
I am really reluctant to do so. Why? A long story from implementing the expand-to-full-image in highlights and also cacorrect leading to a number of added debug logs for checking that. I think the explainations are at least strongly misleading (at least for me) and i would even go as far and remove them all over :-) Just think about: (as in clipping)
```
// 2nd pass: which roi would this operation need as input to fill the given output region?
```
1. What does 2nd pass mean? (We would need to know that the first pass is done before processing the pipe while getting it's dimensions ...)
2. And - we don't care about the output region here at all. 

I think we want/need some sort of dev manual for this - probably in iop_api.h
"While traversing the pixelpipe from last to first module we may modify the input buffer size as required/wanted. Whenever the module is actually processed we will find width x height data as input...." (the rest get's very tricky. Also we would need some info about: "how to process a module if a) data sizes for roi_in/out differ and b) if we have some "position shift" dy/dy are not equal and how does that correspond to size. At least for me that was the **big part** for those modules. 

What do you think?

Reply:
That would certainly be useful too.